+vis/grating.m

```diff
@@ -1,43 +1,47 @@
 function elem = grating(t, grating, window)
 %VIS.GRATING Returns a Signals grating stimulus defining a grating texture
-%  Produces a grating element that is either sinusoidal or square-wave, and
-%  may be windowed by a Gaussian stencil, producing a Gabor patch
+%  Produces a visual element for parameterizing the presentation of a
+%  grating. Produces a grating that can be either sinusoidal or
+%  square-wave, and may be windowed by a Gaussian stencil, producing a 
+%  Gabor patch.
 %
 %  Inputs:
-%    t - any signal with which to derive the network ID.  By convetion we
-%      use the t signal
-%    grating - char array defining the nature of the grating. Options are
-%      'sinusoid' (default) or 'squarewave'
-%    window - char array defining the type of windowing applied.  Options
-%      are 'none' (default) or 'gaussian'
+%    't' - The "time" signal. Used to obtain the Signals network ID.
+%      (Could be any signal within the network - 't' is chosen by
+%      convention).
+%    'grating' - A char array defining the nature of the grating. Options
+%      are 'sinusoid' (default) or 'squarewave'.
+%    'window' - A char array defining the type of windowing applied.
+%      Options are 'gaussian' (default) or 'none'.
 %    
 %  Outputs:
-%    elem - a subscriptable signal containing paramter fields for the
-%      stimulus along with the processed texture layers.  Any parameter may
-%      be a signal.  
+%    'elem' - a subscriptable signal containing fields which parametrize
+%      the stimulus, and a field containing the processed texture layer. 
+%      Any of the fields may be a signal.
 % 
-%  Stimulus (elem) parameters:
-%    grating - see above
-%    window - see above
-%    azimuth - the position of the shape in the azimuth (position of the
-%      centre pixel in visual degrees).  Default 0
-%    altitude - the position of the shape in the altitude. Default 0
-%    sigma - if window is Gaussian, the size of the window in visual 
-%      degrees.  Must be an array of the form [width height].  
+%  Stimulus parameters (fields belonging to 'elem'):
+%    'grating' - see above
```

Comment:
You sure love inverted commas :P Please don't make too many of these stylistic changes, it's a lot of needless work and leaves the door open for the next contribute to comb through the repo changing everything to suite their style, which will probably be different to yours.

Reply:
Eh, I won't comb through the repo making style changes rn, only to files I'm working on. I definitely wouldn't say it's "a lot of needless work" because it certainly helps readability for me and takes little time. It would be nice if style was consistent throughout the repo (*wink*) but I won't try and force that rn

Reply:
To be clear, if there was consistent documentation style across the repo I would gladly adhere to it, and I think most contributors would. I think consistent documentation style is important, and something Chris also has said he wishes was more prevalent throughout Rigbox (and something he says he would've worked on had he had more time - I choose to believe him =D ) Since there isn't, I'm trying to make it consistent in a way that I think is nice, and adheres to a style I've seen multiple other MATLAB programmers use.

Reply:
The MATLAB docs only use inverted commas for the values of variables that are character arrays (e.g. type `help interp1` and `help weboptions`).  All other variables and fieldnames are given without quotes.  

Quotes add a lot of visual noise and it's very rare that a variable name makes the sentence read ambiguously. It is abundantly clear to all readers what the word 'grating' refers to, with or wither quotes.  Here's a bit from the interp1 help with and without quotes:

```
'X' must be a vector. The length of 'X' is equal to 'N'.
If 'V' is a vector, 'V' must have length 'N', and 'Vq' is the same size as 'Xq'.
If 'V' is an array of size ['N','D1','D2',...,'Dk'], then the interpolation is
performed for each 'D1'-by-'D2'-by-...-'Dk' value in 'V(i,:,:,...,:)'. If 'Xq'
is a vector of length 'M', then 'Vq' has size ['M','D1','D2',...,'Dk']. If 'Xq' is 
an array of size ['M1','M2',...,'Mj'], then 'Vq' is of size
['M1','M2',...,'Mj','D1','D2',...,'Dk'].
```
```
X must be a vector. The length of X is equal to N.
If V is a vector, V must have length N, and Vq is the same size as Xq.
If V is an array of size [N,D1,D2,...,Dk], then the interpolation is
performed for each D1-by-D2-by-...-Dk value in V(i,:,:,...,:). If Xq
is a vector of length M, then Vq has size [M,D1,D2,...,Dk]. If Xq is 
an array of size [M1,M2,...,Mj], then Vq is of size
[M1,M2,...,Mj,D1,D2,...,Dk].
```

For me the former is _more_ difficult to read.  Musicians sometimes call it 'fly shit' and like with this, it's up to the composer how much to put into the sheet music.  Sometimes staccato marks and the like are better inferred than written out explicitly.  I think if you plot precision against readability (i.e. how easy it is to sight-read) it doesn't follow a straight line.

When I next edit these function I could change it all back to how it was previously and we'd go back and forth between our two preferred styles.  The other option is to spend ages writing a 20 page document explaining how everything should be formatted.  I'd rather not write a 'Rigbox Style Manual' that details these minutia because no one will read it or follow it.  Instead, I suggest not making these sorts of changes at all.  We talked about how MATLAB's documentation style is pretty inconsistent anyway (although I've yet to see quotes around variable names), so it's not that important to make ours perfectly regular.

Reply:
> I think consistent documentation style is important, and something Chris also has said he wishes was more prevalent throughout Rigbox

But Chris didn't write _any_ documentation, consistent or otherwise :D

Reply:
Well while I may sometimes disagree with your style, I certainly appreciate your work :)

Reply:
Cheers buddy :3